src/BasicBehaveEngine/nodes/world/PlayAnimation.ts

```diff
@@ -0,0 +1,48 @@
+import {BehaveEngineNode, IBehaviourNodeProps} from "../../BehaveEngineNode";
+
+export class PlayAnimation extends BehaveEngineNode {
+    REQUIRED_VALUES = [{id: "animation"}, {id: "speed"}, {id: "loopCount"},  {id: "targetTime"}]
+
+    _animation: number;
+    _speed: number;
+    _loopCount: number;
+    _targetTime: number;
+
+    constructor(props: IBehaviourNodeProps) {
+        super(props);
+        this.name = "PlayAnimation";
+        this.validateValues(this.values);
+        this.validateFlows(this.flows);
+        this.validateConfigurations(this.configuration);
+
+        const {animation, speed, loopCount, targetTime} = this.evaluateAllValues(this.REQUIRED_VALUES.map(val => val.id));
```

Comment:
values should not be evaluated at construction time, general practice is configuration = "class" private variables and the values are temporary variables only evaluated and used during the processNode function

Reply:
Gotcha